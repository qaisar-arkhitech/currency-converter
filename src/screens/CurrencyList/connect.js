import {connect} from "react-redux"

import {selectStyleableTheme} from "../../redux/reducers/themes/selectors"
import {
  selectRatesData,
  selectRatesLoadingError,
  selectRatesLoader,
} from "../../redux/reducers/conversion/selectors"

const mapStateToProps = (state) => ({
  styleableTheme: selectStyleableTheme(state),

  currencies: selectRatesData(state),
  ratesLoadingError: selectRatesLoadingError(state),
  ratesLoader: selectRatesLoader(state),
})

const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)
