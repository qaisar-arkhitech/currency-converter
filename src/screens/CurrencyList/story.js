import React from "react"
import CurrencyList from "."

export default {
  title: "CurrencyList",
  storyFn() {
    return (
      <div>
        <CurrencyList />
      </div>
    )
  },
}
