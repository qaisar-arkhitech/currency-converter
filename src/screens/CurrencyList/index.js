import React from "react"
import flow from "lodash/flow"
import {View, Text, StatusBar, FlatList} from "react-native"
import PropTypes from "prop-types"
import themeStyles from "./styles"
import {RowItem, RowSeparator} from "../../components/RowItem"
import connect from "./connect"

// CurrencyList Component content
export const CurrencyList = ({navigation, currencies, styleableTheme}) => {
  const styles = themeStyles(styleableTheme)
  return (
    <View style={styles.root}>
      <StatusBar barStyle="dark-content" />
      <FlatList
        data={currencies}
        renderItem={({item}) => {
          return (
            <RowItem
              title={item.name}
              onPress={() => {
                navigation.pop()
              }}
              rightIcon={null}
            />
          )
        }}
        keyExtractor={(item) => item.name}
        ItemSeparatorComponent={() => <RowSeparator />}
        ListFooterComponent={() => <View />}
      />
    </View>
  )
}

// CurrencyList Proptypes
CurrencyList.propTypes = {
  currencies: PropTypes.object,
  navigation: PropTypes.object.isRequired,
}

// CurrencyList Default props
CurrencyList.defaultProps = {
  currencies: {},
}

export default flow([connect])(CurrencyList)
