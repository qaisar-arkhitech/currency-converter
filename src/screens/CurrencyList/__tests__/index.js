import React from "react"
import renderer from "react-test-renderer"
import CurrencyList from ".."

const createProps = () => ({
  classes: {},
})

describe("<CurrencyList />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<CurrencyList {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
