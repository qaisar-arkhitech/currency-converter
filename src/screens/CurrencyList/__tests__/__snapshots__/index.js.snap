// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyList /> Does not explode 1`] = `
<View>
  <Text>
    CurrencyList
  </Text>
</View>
`;
